enhancements/ingress/aws-elb-idle-timeout.md

```diff
@@ -0,0 +1,328 @@
+
+1. Introduce a platform-agnostic API (which would be impossible to implement on most platforms).
+1. Introduce an equivalent API for other platforms or LB types besides AWS Classic ELBs.
+
+## Proposal
+
+To enable the cluster administrator to configure the connection timeout period
+for AWS Classic ELBs, the IngressController API is extended by adding an
+optional `ConnectionIdleTimeout` field with type `*metav1.Duration` to
+`AWSClassicLoadBalancerParameters`:
+
+```go
+// AWSClassicLoadBalancerParameters holds configuration parameters for an
+// AWS Classic load balancer.
+type AWSClassicLoadBalancerParameters struct {
+	// connectionIdleTimeout specifies the maximum time period that a
+	// connection may be idle before the load balancer closes the
+	// connection.  The value must be parseable as a time duration value;
+	// see <https://pkg.go.dev/time#ParseDuration>.  The default value for
+	// this field is 60s.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Format=duration
+	// +optional
+	ConnectionIdleTimeout *metav1.Duration `json:"connectionIdleTimeout"`
+}
+```
+
+The following example configures a 2-minute timeout period:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      providerParameters:
+        type: AWS
+        aws:
+          type: Classic
+          classicLoadBalancer:
+            connectionIdleTimeout: 2m
+```
+
+### Validation
+
+Omitting `spec.endpointPublishingStrategy`,
+`spec.endpointPublishingStrategy.loadBalancer`,
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters`,
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws`,
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.classicLoadBalancer`,
+or
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.classicLoadBalancer.connectionIdleTimeout`
+is valid and specifies the default behavior.  The API validates that any
+provided value is a time duration value.
+
+### User Stories
+
+#### As a cluster administrator, I need my IngressController's ELB to have a 2-minute timeout
+
+To satisfy this use-case, the cluster administrator can set the
+IngressController's
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.classicLoadBalancer.connectionIdleTimeout`
+field to `2m`.
+
+#### As an project administrator, I need my application to have a 5-minute timeout
+
+To satisfy this use-case, the cluster administrator must set the
+IngressController's
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.classicLoadBalancer.connectionIdleTimeout`
+field to `5m` or longer, and the project administrator must similarly set a
+Route annotation for the application's Route:
+
+```shell
+oc patch -n my-project routes/my-route haproxy.router.openshift.io/timeout=5m
```

Comment:
this is supposed to be the annotate comment right? patch doesn't work like this.